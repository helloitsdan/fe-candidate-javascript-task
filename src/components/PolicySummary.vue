<template>
  <div>
    <h3 class="is-marginless">{{ insured_entities | summariseInsuredEntities }}</h3>
    <p v-if="status === 'ACTIVE'"><strong>Active</strong></p>

    <p>Started on {{ start_date | moment("dddd, MMMM Do YYYY") }}</p>
  </div>
</template>

<script lang="ts">
import Vue from "vue";

export default Vue.extend({
  name: "PolicySummary",
  props: {
    uuid: String,
    status: String,
    insured_entities: Array,
    start_date: String
  }
});
</script>
