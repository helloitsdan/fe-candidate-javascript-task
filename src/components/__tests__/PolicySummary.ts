import { shallowMount } from "@vue/test-utils";

import PolicySummary from "../PolicySummary.vue";

describe("PolicySummary", () => {
  const fakePolicy = {
    uuid: "1234",
    status: "ACTIVE",
    start_date: "2016-07-13T02:57:55.235Z",
    insured_entities: [
      { species: "CAT" },
      { species: "CAT" },
      { species: "DOG" }
    ],
  };

  it("renders", () => {
    const wrapper = shallowMount(PolicySummary, {
      propsData: fakePolicy,
    });

    expect(wrapper.element).toMatchSnapshot();
  });
});
