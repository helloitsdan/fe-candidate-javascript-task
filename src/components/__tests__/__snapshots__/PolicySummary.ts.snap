// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PolicySummary renders 1`] = `
<div>
  <h3
    class="is-marginless"
  >
    2 cats, 1 dog
  </h3>
   
  <p>
    <strong>
      Active
    </strong>
  </p>
   
  <p>
    Started on Wednesday, July 13th 2016
  </p>
</div>
`;
